### PR TITLE
CRAYSAT-1571: Add workflows for spell check, link check and lint check.

### PR DESCRIPTION
**Pull request by ekamin-hpe**
## Summary and Scope

This commit adds links 

## Issues and Related PRs

* Resolves [CRAYSAT-1571](https://jira-pro.its.hpecorp.net:8443/browse/CRAYSAT-1571)

## Testing

To test this, I ran each of the containers against all the files in this repository. See the following gist for output: https://gist.github.hpe.com/eli-kamin/9607d3c79245ca44f301cb0420ccff7b

Note that the workflows being introduced only test against modified files.

## Pull Request Checklist

- [X] Version number(s) incremented, if applicable
- [X] Copyrights updated
- [X] License file intact
- [X] Target branch correct
- [X] CHANGELOG.md updated
- [X] Testing is appropriate and complete, if applicable
- [X] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: rpm/README.md [hunk 1]
```diff
@@ -21,23 +21,23 @@ and enhancements that were not available until after the software release.
...
+   ```bash
    rpm -q docs-sat
    ```
 
 2. Download and upgrade the latest documentation RPM.
 
    For example, to download and upgrade to the latest SAT 2.4 documentation RPM:
 
-   ```
+   ```bash
```
Comment (ekamin-hpe) [inline rpm/README.md:32]: These look fine with `bash` highlighting though I'm thinking for consistency they should be `screen`.

File: docs/install.md [hunk 2]
```diff
@@ -529,21 +529,22 @@ The following is the procedure to globally configure the username used by SAT an
...
-1.  Edit `~/.config/sat/sat.toml` and set the username option in the `api_gateway` section of the config file. E.g.:
+1.  Edit `~/.config/sat/sat.toml` and set the username option in the `api_gateway` section of the config file. For
+    example:
 
     ```screen
     username = "crayadmin"
     ```
 
-1.  Run `sat auth`. Enter your password when prompted. E.g.:
+1.  Run `sat auth`. Enter your password when prompted.
```
Comment (haasken-hpe) [inline docs/install.md]: Still feels like a "For example:" makes sense given that the username is an example.

File: docs/install.md [hunk 3]
```diff
@@ -475,24 +475,24 @@ Below is a table describing SAT commands and the types of authentication they re
...
-|`sat bootsys`|Requires authentication to the API gateway. Requires kubernetes configuration and authentication, which is configured on `ncn-m001` during the install. Some stages require passwordless SSH to be configured to all other NCNs. Requires S3 to be configured for some stages.|`sat-bootsys`|Boot or shutdown the system, including compute nodes, application nodes, and non-compute nodes (NCNs) running the management software.|
+|`sat bootprep`|Requires authentication to the API gateway. Requires Kubernetes configuration and authentication, which is done on `ncn-m001` during the install.|`sat-bootprep`|Prepare to boot nodes with images and configurations.|
+|`sat bootsys`|Requires authentication to the API gateway. Requires Kubernetes configuration and authentication, which is configured on `ncn-m001` during the install. Some stages require password-less SSH to be configured to all other NCNs. Requires S3 to be configured for some stages.|`sat-bootsys`|Boot or shutdown the system, including compute nodes, application nodes, and non-compute nodes (NCNs) running the management software.|
```
Comment (haasken-hpe) [inline docs/install.md:481]: I'd consider adding "passwordless" to the `.spelling` file as an exception instead.

File: docs/install.md [hunk 4]
```diff
@@ -529,21 +529,22 @@ The following is the procedure to globally configure the username used by SAT an
...
+1.  Run `sat auth`. Enter your password when prompted.
 
     ```screen
     ncn-m001# sat auth
     Password for crayadmin:
     Succeeded!
     ```
 
-1.  Other `sat` commands are now authenticated to make requests to the API gateway. E.g.:
+1.  Other `sat` commands are now authenticated to make requests to the API gateway.
```
Comment (haasken-hpe) [inline docs/install.md]: "For example:" also still seems like it makes sense here.

File: docs/introduction.md [hunk 5]
```diff
@@ -52,8 +52,8 @@ have their own set of options.
 
 ### SAT Container Environment
 
-The `sat` command-line utility runs in a container using podman, a daemonless container runtime. SAT runs on Kubernetes
-manager nodes. A few important points about the SAT container environment include the following:
+The `sat` command-line utility runs in a container using Podman, a daemon-less container runtime. SAT runs on
```
Comment (haasken-hpe) [inline docs/introduction.md:55]: given that Podman's website uses "daemonless", this is also one I'd consider adding and exception for.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -26,7 +26,7 @@ Contributors must follow the guidelines below:
    separate commits addressing feedback is that it provides an easy way for
    reviewers to see only what changed if desired.
 
-# Copying
+## Copying
```
Comment (haasken-hpe) [inline CONTRIBUTING.md:29]: Why did linting want this change?

File: docs/install.md [hunk 7]
```diff
@@ -802,15 +804,15 @@ the new default behavior.
 The following commands trigger messages that have been changed from `stdout`
 print calls to `INFO`-level (or `WARNING`- or `ERROR`-level) log messages:
 
-```
+```screen
```
Comment (haasken-hpe) [inline docs/install.md:807]: Seems like this block and the next one would make more sense as a list of fixed-width items.

File: .github/scripts/check_quotes.sh [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# Note: this script was copied from docs-csm:
+# https://github.com/Cray-HPE/docs-csm/blob/05236b40427b46fc837674967891d9d5e754ae73/runLint.sh
+function check_quotes {
+    local error=0
+    printf "=============== Linting \” (https://www.compart.com/en/unicode/U+201C and U+201D) ... \n"
+    grep -n -R \” *.md && echo >&2 'Malformed quotes detected (bad: ” vs. good: ").' && error=1
```
Comment (haasken-hpe) [inline .github/scripts/check_quotes.sh:30]: I'm just curious, why does this check only look for the right double quotation mark (U+201D) and not the left double quotation mark (U+201C)?

File: .github/workflows/markdown-lint.yaml [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+      uses: tj-actions/changed-files@v18.7
+      with:
+        files: "**/*.md"
+        files_ignore: ".github/**/*"
+
+    - uses: docker://ghcr.io/igorshubovych/markdownlint-cli:v0.31.1
+      with:
+        args: "--config .github/config/markdown_style.yaml ${{ steps.changed-files.outputs.all_changed_files }}"
+
+    - name: Check for illegal ASCII quote characters
```
Comment (haasken-hpe) [inline .github/workflows/markdown-lint.yaml:46]: The name of this step is a little misleading. I think "Check for illegal double quotation mark Unicode characters" would make more sense.

**Review by haasken-hpe — APPROVED** [review]
Very minor comments.

File: CONTRIBUTING.md (hunk 6 above)
Comment (ekamin-hpe) [reply CONTRIBUTING.md:29]: Multiple top-level headings in the same document: https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md025---multiple-top-level-headings-in-the-same-document

File: .github/scripts/check_quotes.sh (hunk 8 above)
Comment (ekamin-hpe) [reply .github/scripts/check_quotes.sh:30]: I'm not sure. https://github.com/Cray-HPE/docs-csm/commit/05236b40427b46fc837674967891d9d5e754ae73 seems to have changed the print message to say that it was checking for both characters, but the `grep` command didn't change. It looks pretty simple to fix.

Comment (ekamin-hpe) [reply .github/scripts/check_quotes.sh:30]: I'm also noticing now that it uses recursive `grep` but specifies the glob pattern `*.md`, which as a result only scans markdown files in the top-level directory. We ought to fix that, imo.

Comment (haasken-hpe) [reply .github/scripts/check_quotes.sh]: Yeah, makes sense to fix

File: CONTRIBUTING.md (hunk 6 above)
Comment (haasken-hpe) [reply CONTRIBUTING.md:29]: Oh, that's kind of a weird one

File: .github/scripts/check_quotes.sh (hunk 8 above)
Comment (ekamin-hpe) [reply .github/scripts/check_quotes.sh]: Also submitting https://github.com/Cray-HPE/docs-csm/pull/2533

**Pull request merged**